main.py

```diff
@@ -1,74 +1,77 @@
 #!/usr/bin/python3
 
-import time, sys
-from rgbmatrix import graphics, RGBMatrix, RGBMatrixOptions
-from train_updater import get_next_trains
-import logging
-
-# set up logging
-logging.basicConfig(filename="main.log", level=logging.DEBUG)
-
-# drawing 10x10 L train logo
-def draw_l_train_logo(x, y, canvas):
-    for j in range(3):
-        offset = j + 1
-        for i in range(offset, 10 - offset):
-            canvas.SetPixel(x + i, y + 2 - j, 117, 119, 122)
-            canvas.SetPixel(x + i, y + 8 + j, 117, 119, 122)
-    for j in range(3, 8):
-        for i in range(10):
-            if i == 4:
-                canvas.SetPixel(x + i, y + j, 255, 255, 255)
-            elif j == 7 and (i == 5 or i == 6):
-                canvas.SetPixel(x + i, y + j, 255, 255, 255)
-            else:
-                canvas.SetPixel(x + i, y + j, 117, 119, 122)
-
-def run():
-    logging.info("Starting l-train-display.")
-
-    options = RGBMatrixOptions()
-    options.rows = 32
-    options.cols = 64
-    options.chain_length = 1
-    options.parallel = 1
-    options.hardware_mapping = "adafruit-hat"
-
-    matrix = RGBMatrix(options=options)
-
-    dir_font = graphics.Font()
-    dir_font.LoadFont("5x8.bdf")
-    time_font = graphics.Font()
-    time_font.LoadFont("5x8.bdf")
-
-    l_train_color = graphics.Color(167, 169, 172)
-    font_color = graphics.Color(250, 200, 50)
-
-    canvas = matrix.CreateFrameCanvas()
-
+import time, sys, threading
+from train_updater import TrainUpdater, Trip, Direction
+from train_displayer import TrainDisplayer
+import mta_api_key
+import constants
+
+updater = TrainUpdater(constants.MTA_API_URL, mta_api_key.key)
+displayer = TrainDisplayer("5x8.bdf")
+
+lock = threading.Lock()
+next_trains_trip = {
+    "North": None,
+    "South": None
+}
+
+def parse_arrival_time(arrival):
+    now = time.time()
+    arrival_in_min = round((arrival - now)/60)
+    if arrival_in_min < -5:
+        return "-"
+    else:
+        return str(arrival_in_min)
+
+def parse_trips_and_update_display():
     while True:
-        manhattan_in, brooklyn_in = get_next_trains()
-        canvas.Clear()
+        lock.acquire()
+        trip_north = next_trains_trip["North"]
+        trip_south = next_trains_trip["South"]
+        lock.release()
```

Comment:
It may be safer to use the lock from a with statement instead of manually acquiring and releasing

```suggestion
        with lock:
            trip_north = next_trains_trip["North"]
            trip_south = next_trains_trip["South"]
```